### ISSUE TITLE
Recover WebSocket Connections when closing

### ISSUE THREAD

**Issue by adrum** [lukas-hetzenecker/home-assistant-remote#3]
After sometime, my connection appears to be stuck closing. Any time I try to control the remote instance from my main instance, I receive the following error message:

```
[36mhomeassistant_1  |[0m 2018-12-21 13:03:29 WARNING (MainThread) [aiohttp.websocket] websocket connection is closing.
[36mhomeassistant_1  |[0m 2018-12-21 13:03:29 ERROR (MainThread) [homeassistant.core] Error doing job: Task exception was never retrieved
[36mhomeassistant_1  |[0m Traceback (most recent call last):
[36mhomeassistant_1  |[0m   File "/config/custom_components/remote_homeassistant.py", line 235, in forward_event
[36mhomeassistant_1  |[0m     await self._connection.send_json(data)
[36mhomeassistant_1  |[0m   File "/usr/local/lib/python3.6/site-packages/aiohttp/client_ws.py", line 137, in send_json
[36mhomeassistant_1  |[0m     await self.send_str(dumps(data), compress=compress)
[36mhomeassistant_1  |[0m   File "/usr/local/lib/python3.6/site-packages/aiohttp/client_ws.py", line 125, in send_str
[36mhomeassistant_1  |[0m     await self._writer.send(data, binary=False, compress=compress)
[36mhomeassistant_1  |[0m   File "/usr/local/lib/python3.6/site-packages/aiohttp/http_websocket.py", line 630, in send
[36mhomeassistant_1  |[0m     return await self._send_frame(message, WSMsgType.TEXT, compress)
[36mhomeassistant_1  |[0m   File "/usr/local/lib/python3.6/site-packages/aiohttp/http_websocket.py", line 596, in _send_frame
[36mhomeassistant_1  |[0m     self.transport.write(header + mask + message)
[36mhomeassistant_1  |[0m   File "uvloop/handles/stream.pyx", line 671, in uvloop.loop.UVStream.write
[36mhomeassistant_1  |[0m   File "uvloop/handles/handle.pyx", line 159, in uvloop.loop.UVHandle._ensure_alive
[36mhomeassistant_1  |[0m RuntimeError: unable to perform operation on <TCPTransport closed=True reading=False 0x557207f2a308>; the handler is closed
```

I've added some comments to the top of my custom component file so line 235 on my install is 229 in the repo.

**Comment by davemuench** [comment]
I am running into this as well FWIW, haven't had a chance to dig into it yet.

**Comment by lukas-hetzenecker** [comment]
Fixed in c2474103af31e46eba74447db34745795e4f4d50

**Comment by adrum** [comment]
Excellent! Thank you.
